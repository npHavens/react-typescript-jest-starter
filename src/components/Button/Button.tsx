
import React, { useState, SFC } from "react";
import styled from 'styled-components';
import styledComponentsTS from 'styled-components-ts';
import { ButtonProps } from '../../declarations/Button';

const StyledButton: SFC<ButtonProps> = () => {
  const [active, setActive] = useState(false);

  const Button = styledComponentsTS<ButtonProps>(styled.button)`
    background: ${active ? 'red' : 'black'};
    color: #fff;
    border: 0;
    padding: 7px 10px;`;

  return (
    <div className="Button">
      <Button onClick={() => setActive(!active)}>Submit</Button>
    </div>
  );

};

export default StyledButton;