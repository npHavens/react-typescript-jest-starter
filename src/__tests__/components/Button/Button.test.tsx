import React, { Fragment } from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';


import Button from '../../../components/Button/Button';

import * as renderer from 'react-test-renderer';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
    .create(<Button/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
