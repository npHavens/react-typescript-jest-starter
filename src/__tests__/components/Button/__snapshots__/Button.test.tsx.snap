// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  background: black;
  color: #fff;
  border: 0;
  padding: 7px 10px;
}

<div
  className="Button"
>
  <button
    className="c0"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
